front-end/components/appInputComponents/ChatTextField.tsx

```diff
@@ -0,0 +1,90 @@
+import { useState, ChangeEvent, KeyboardEvent } from 'react';
+import Box from '@mui/joy/Box';
+import Textarea from '@mui/joy/Textarea';
+import Button from '@mui/material/Button';
+import Stack from '@mui/material/Stack';
+import SendIcon from '@mui/icons-material/Send';
+
+import axios from 'axios';
+
+export default function ChatTextField(){
+  
+    const submitStyle = {
+      position: 'fixed',
+      bottom: '80px',
+      right: '80px'
+    }
+    
+  const [initialValue, setValue] = useState('');
+
+    //Sending Data to the chatGPT API
+    const sendingData = async() => {
+      try{
+        const res = await axios.post('http://localhost:8080/chat_input', {
+          initialValue
+        });
+        console.log(res.data);
+      }catch(e){
+        console.log(e);
+      }
+      setValue('');
+    }
+
+    //User Input
+  const userInput = (e: ChangeEvent<HTMLTextAreaElement>) => {
+    setValue(e.target.value);
+  }
+
+  //Submit Input with 'Enter' key-word
+  function enterSubmit(e: KeyboardEvent){
+    if(document.activeElement === e.target){
+      if(e.key === 'Enter'){
+        e.preventDefault();
+        sendingData();
+      } 
+    }
+  }
+
+  //Submit Input with 'Send' button
+  const submitChat = (event: { preventDefault: () => void; }) => {
+    event.preventDefault();
+    sendingData();
+  }
+
+  //Render the Chat Box
+  return (
+    <Box
+      sx={{
+        py: 2,
+        display: 'grid',
+        gap: 2,
+        alignItems: 'center',
+        flexWrap: 'wrap',
+      }}
+    >
+    <form  onSubmit={submitChat}>
+      <Stack 
+        direction="row" 
+        spacing={1} 
+        sx={submitStyle}>
+      <Textarea 
+        name="Outlined" 
+        placeholder="Chat Box" 
+        variant="outlined" 
+        onChange={userInput} 
+        value={initialValue} 
+        sx={{width: 250}} 
+        onKeyDown={enterSubmit} 
+      />
+      <Button 
```

Comment:
If you want to have the chatbox only show after button is clicked. Try popover: https://mui.com/material-ui/react-popover/.

The general idea is to have a `const [open, setOpen] = useState(false)` that the button changes and will render the chatbox if the `open` is true

Reply:
This was to make the enter button send data to the API

Reply:
I will look at the popover on mui